File: update_readme.py
import sys
import os

def update_readme(added_files):
    if not added_files:
        return

    for added_file in added_files.split(','):
        if added_file.endswith('.md'):
            date, title = added_file.split("_", 1)
            title = title.replace(".md", "")
            new_link = f"- [[{date}] {title}](https://github.com/kimbongjune/TIL/blob/main/{added_file})\n"
            
            with open("README.md", "a") as f:
                f.write(new_link)

if __name__ == "__main__":
    added_files = sys.argv[1]
    update_readme(added_files)

File: .github/workflows/scripts/update_readme.js
const fs = require('fs');
const path = require('path');
const { Octokit } = require("@octokit/rest");

const main = async () => {
  const octokit = new Octokit();
  const readmePath = path.join(__dirname, '../../README.md');
  let readmeContent = fs.existsSync(readmePath) ? fs.readFileSync(readmePath, 'utf8') : '# Today I Learned(TIL)\n## 카테고리';

  const { data: commits } = await octokit.repos.listCommits({
    owner: process.env.GITHUB_REPOSITORY.split('/')[0],
    repo: process.env.GITHUB_REPOSITORY.split('/')[1]
  });

  const changedFiles = commits[0].files;
  const changedMdFiles = changedFiles.filter(file => file.filename.endsWith('.md') && file.filename !== 'README.md');

  changedMdFiles.forEach(file => {
    const isDeletedInMainBranch = file.status === 'removed';
    const fileName = file.filename;
    const date = fileName.split('_')[0];
    const title = fileName.split('_')[1].replace('.md', '').replace(/ /g, '%20');
    const link = `- [[${date}] ${title}](https://github.com/${process.env.GITHUB_REPOSITORY}/blob/main/${fileName})`;

    if (isDeletedInMainBranch) {
      if (readmeContent.includes(link)) {
        readmeContent = readmeContent.replace(link, '');
      }
    } else {
      if (!readmeContent.includes(link)) {
        readmeContent += '\n' + link;
      }
    }
  });

  fs.writeFileSync(readmePath, readmeContent);
};

main();
